src/zip.h

```diff
@@ -331,7 +331,28 @@ extern int zip_extract_stream(const char *stream, size_t size, const char *dir,
  *
  * @return the zip archive handler or NULL on error
  */
-extern struct zip_t *zip_open_stream(const char *stream, size_t size);
+extern struct zip_t *zip_open_stream(const char *stream, size_t size,
```

Comment:
I think it can be confusing to use the _stream_ API.
I mean `(NULL, 0, ...)` for write etc.
I propose to split this function into two function.

So, generally I thought about the _stream/in memory_ API.
And I think we should go (for consistency) with something like:
```c
zip_open_rstream // to open for read
zip_open_wstream // to open for write

zip_copy_stream // to get the copy of stream

zip_extract_stream // this is what we already have

zip_close_stream // I wonder if we need it (maybe for consistency), because I think it's the same as zip_close
```

BTW., I don't like the name stream, but frankly I don't know what will work better and let us define intuitive API. 
Neither `inmemory` nor `archive` sound good in our case.

Reply:
I think zip_open_stream( ) should just like zip_open( ). 
Here's how to use them https://github.com/kuba--/zip/blob/9d59e214651468161bb1fbea42f1de6204bc732e/test/test.c#L551 to https://github.com/kuba--/zip/blob/9d59e214651468161bb1fbea42f1de6204bc732e/test/test.c#L561
zip_extract_stream ( ) this interface unzips the stream to a file (disk).
you can check this [function](https://github.com/kuba--/zip/blob/9d59e214651468161bb1fbea42f1de6204bc732e/test/test.c#L546) to get more information.

Reply:
Regarding `zip_open_stream` - I know what you mean, but this function takes 4 parameters, but always only 2 are valid. For `zip_open` only level can be ignored when you open in read mode.
Don't get me wrong - I'd like to have just one function, but my main concern is for writing, we will have the first two params. optional:
```
(NULL, 0, ZIP_DEFAULT_COMPRESSION_LEVEL, 'w')
```
but for reading only level:
```
(buf_encode, n, 0, 'r')
```

Anyway, I'll double think about it.

Regarding `zip_extract_stream` - yeah, I know. I just wanted to list all stream-like functions here, to see if we are consistent with naming.

Reply:
I understand your concerns but this increases the number of APIs.
And consider the use with zip_close_stream(). The zip_open_stream() name is more reasonable. BTW, zip_close_stream() is not the same as zip_close(). zip_close() calls zip_write_end() to write the end of centerdir at the end of the file, but zip_close_stream() does not call zip_write_end(). Because it has been done in zip_stream_cpy().

Reply:
But if you insist, I will modify it. : )

Reply:
Ok, lets keep the function signature, but we need to document the convention - how to open for read and how to open for write